Dependency upgrades and moving to dep.

crspeller opened the pull request:
- Now I am become Death, the destroyer of worlds. (because dep prunes the downloaded repositories for unneeded files)
- Everything you need to know about dep: https://golang.github.io/dep/docs/daily-dep.html

jwilander commented:
I'm jealous of all those deletions 😢

lieut-data commented:
@jwilander, yeah but just think how much code @crspeller will have to write to start making a net contribution again ;P

```
> git log --author="crspeller" --pretty=tformat: --numstat | awk '{ add += $1; subs += $2; loc += $1 - $2 } END { printf "added lines: %s, removed lines: %s, total lines: %s\n", add, subs, loc }' -
added lines: 2046326, removed lines: 4263795, total lines: -2217469
```

crspeller commented:
@lieut-data No no no, your math is wrong. it's:
```
git log --author="crspeller" --pretty=tformat: --numstat | awk '{ add += $1; subs += $2; loc += $1 + $2 } END { printf "added lines: %s, removed lines: %s, total lines: %s\n", add, subs, loc }' -
added lines: 2046326, removed lines: 4263795, total lines: 6310121
```
😄
